spaghetto/src/Spago/Psa/Printer/Default.purs

```diff
@@ -0,0 +1,275 @@
+-- A majority of this code was copied from
+-- - https://github.com/natefaubion/purescript-psa-utils
+-- 
+-- To fullfil license requirements
+--   Copyright © Nathan Faubion
+--   https://opensource.org/license/mit/
+module Spago.Psa.Printer.Default
+  ( renderWarning
+  , renderError
+  , renderStats
+  , renderVerboseStats
+  , print
+  ) where
+
+import Prelude
+
+import Ansi.Codes as Ansi
+import Control.Alternative as Alternative
+import Data.Array as Array
+import Data.Foldable (fold, foldMap, maximum, maximumBy)
+import Data.FoldableWithIndex (forWithIndex_)
+import Data.Maybe (Maybe(..), fromMaybe, maybe)
+import Data.Monoid (power)
+import Data.String as Str
+import Data.Tuple (Tuple(..), uncurry)
+import Data.Unfoldable (unfoldr)
+import Dodo as D
+import Dodo.Ansi as DA
+import Effect (Effect)
+import Effect.Console as Console
+import Foreign.Object as FO
+import Spago.Core.Config as Core
+import Spago.Psa.Output (OutputStats, Output)
+import Spago.Psa.Types (Lines, Position, PsaAnnotedError, PsaOutputOptions, PsaPath(..))
+
+-- | Prints output to the console.
+print :: PsaOutputOptions -> Output -> Effect Unit
+print options output = do
+  forWithIndex_ output.warnings \i warning -> do
+    Console.error $ printDoc (renderWarning lenWarnings (i + 1) warning)
+    Console.error ""
+
+  forWithIndex_ output.errors \i error -> do
+    Console.error $ printDoc (renderError lenErrors (i + 1) error)
+    Console.error ""
+
+  Console.error $ printDoc (renderStats' output.stats)
+
+  where
+  printDoc
+    | options.color = D.print DA.ansiGraphics D.twoSpaces
+    | otherwise = D.print D.plainText D.twoSpaces
+  lenWarnings = Array.length output.warnings
+  lenErrors = Array.length output.errors
+  renderStats' = case options.statVerbosity of
+    Core.NoStats -> mempty
+    Core.CompactStats -> renderStats
+    Core.VerboseStats -> renderVerboseStats
+
+renderWarning :: Int -> Int -> PsaAnnotedError -> D.Doc Ansi.GraphicsParam
+renderWarning = renderWrapper Ansi.BrightYellow
+
+renderError :: Int -> Int -> PsaAnnotedError -> D.Doc Ansi.GraphicsParam
+renderError = renderWrapper Ansi.BrightRed
```

Comment:
Should we add a dark/light theme option that changes whether normal Red/Yellow or bright Red/Yellow are used? This works for my terminal colors but may not for others.

Reply:
I can try this out on a few terminals and report. But I'd say that if the colors are wrong it's likely that we'll get bug reports quickly, as this is going to be the default behaviour 🙂